packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h

```diff
@@ -112,6 +114,6 @@ inline static UIColor *RCTUIColorFromSharedColor(const facebook::react::SharedCo
     return [UIColor whiteColor];
   }
 
-  auto components = facebook::react::colorComponentsFromColor(sharedColor);
-  return [UIColor colorWithRed:components.red green:components.green blue:components.blue alpha:components.alpha];
+  facebook::react::Color color = *sharedColor;
+  return (UIColor *)facebook::react::unwrapManagedObject(color.getUIColor());
```

Comment:
Color is now an (somewhat more) expensive struct instead of just an int. We should avoid copying: SharedColor's `operator*` returns a value and not a reference (which maybe is also something we can fix).

Given that we always allocate a UIColor, the shortcuts above in terms of return clear/black/white may no longer be necessary.